### PR TITLE
UI improvements

### PR DESCRIPTION
**Pull request by DavidBachmann**
## TuxFab (Tux Floating Action Button)
**Normal**
image
**Hovered**
image
**Items hovered**
image
**Active**
image

### Sidebar update
screen shot 2017-03-22 at 15 25 25

### Buttons
**Normal**
image
**Hover**
image

### Inputs
**Normal**
image
**Animates on focus**
image
**Same with TagEditor**
image

### CODE REVIEW

**Comment by DavidBachmann** [comment]
Extracted buttons into Button component that's inspired by material (you're welcome @eirikurn and @rthor)

**Normal**
image

**Hover**
image

**Comment by eirikurn** [comment]
Sweet!

I like the form buttons.

Good job with the fab, but we should rethink what functions we want inside it (they should be primary actions, consistent and have obvious icons), and where to move stuff that doesn't fit in the fab.

File: packages/tux/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+export interface Button {
+  type: string
+  themeColor: string
+  flat: boolean
+  onClick: (e: React.SyntheticEvent<any>) => void
+  children: any
+}
+
+
+const Button = ({ type, flat, themeColor, raised, onClick, children }: Button) => {
```
Comment (eirikurn) [inline packages/tux/src/components/Button/Button.tsx]: `onClick` isn't used.

File: packages/tux/src/components/AlertBar/index.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react'
 import classNames from 'classnames'
-import { tuxColors } from '../../styles'
+import { tuxColors } from '../../colors'
```
Comment (eirikurn) [inline packages/tux/src/components/AlertBar/index.tsx]: We should call this a `theme`, with a few semantic primary variables which apply to component specific variables. Something like:

```
export const Theme = {
  primary: 'pink',
  secondary: 'green',
  foreground: '#333',
  background: '#fff',
  altBackground: '#ddd',
  border: '#666',
}

export const Button = {
  primaryBG: Theme.primary,
  primaryBGHover: fade(Theme.primary, 0.9),
  secondaryBG: Theme.background,
  secondaryBGHover: Theme.altBackground,
  ...
}
```

Happy to discuss more or do this later.

Comment (eirikurn) [reply packages/tux/src/components/AlertBar/index.tsx]: then later we can make helpers to edit this theme, either by changing main theme variables or going deeper.

File: packages/tux/src/components/TuxFab/TuxFab.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            onClick={onClickEdit}>
+              <i className={isEditing ? activeClassName : inActiveClassName}></i>
+          </a>
+          <a className="TuxFab-item" href="/"><i className="icon icon-white_question"></i></a>
+          <a className="TuxFab-item" href="/"><i className="icon icon-git"></i></a>
+          <a
+            onClick={this.login}
+            className="TuxFab-item TuxFab-signInOut">
+            <i className="icon icon-user"></i>
+          </a>
```
Comment (eirikurn) [inline packages/tux/src/components/TuxFab/TuxFab.tsx]: Don't need to do this now, but if we end up using this, it should display the button label on hover and be optimised for touch.

**Review by eirikurn — CHANGES_REQUESTED** [review]
Sweet. One bug in `Button`.

**Comment by eirikurn** [comment]
So I'm thinking that the FAB should have this primary action to toggle edit mode for the page. It would need some other icon, as the + is used for creating new things.

I imagine using secondary actions to add specific models. "Add Page", "Add Gallery Item", "Add News", and maybe a settings button to hide the rest in a modal.

My issues are:

* How does the user login. I haven't seen a login button in a FAB before.
* Having a FAB for user-provided actions (like "Create <model>") involves more work to give each fab a meaningful icon or background. This might be difficult for some models/users. Would a series of "+" actions with different labels work?

**Comment by DavidBachmann** [comment]
Todo: When in admin mode and **not** logged in TuxFab should only be a log-in action.
Haven't been able to display the Fab when I'm not logged in.

**Pull request merged**